### PR TITLE
Add checker of CommiteeHash for Tx/FB/VC/Micro blocks

### PR DESCRIPTION
**Pull request by kaikawaliu**
## Description

Add checker of CommiteeHash for Tx/FB/VC/Micro blocks, which used to be dummy value.

### Implementation

- [X] **ready for review**

### Integration Test (Core Team)

- [x] local machine test

- [x] 200 nodes cloud test

### CODE REVIEW

File: src/libDirectoryService/FinalBlockPreProcessing.cpp [hunk 1]
```diff
@@ -164,13 +164,21 @@ void DirectoryService::ComposeFinalBlock() {
 
   StateHash stateRoot = AccountStore::GetInstance().GetStateRootHash();
 
+  // Compute the CommitteeHash member of the BlockHeaderBase
+  CommitteeHash committeeHash;
+  if (!Messenger::GetDSCommitteeHash(*m_mediator.m_DSCommittee,
+                                     committeeHash)) {
+    LOG_EPOCH(WARNING, to_string(m_mediator.m_currentEpochNum).c_str(),
+              "Messenger::GetDSCommitteeHash failed.");
```
Comment (ansnunez) [inline src/libDirectoryService/FinalBlockPreProcessing.cpp]: Return `false` here? (Like what is done for DS Block announcement)

**Comment by bb111189** [comment]
Add PR description

File: src/libDirectoryService/ViewChangePreProcessing.cpp [hunk 2]
```diff
@@ -261,6 +279,14 @@ void DirectoryService::ComputeNewCandidateLeader() {
         m_mediator.m_DSCommittee->at(m_viewChangeCounter).second;
   }
 
+  // Compute the CommitteeHash member of the BlockHeaderBase
+  CommitteeHash committeeHash;
+  if (!Messenger::GetDSCommitteeHash(*m_mediator.m_DSCommittee,
+                                     committeeHash)) {
+    LOG_EPOCH(WARNING, to_string(m_mediator.m_currentEpochNum).c_str(),
+              "Messenger::GetDSCommitteeHash failed.");
```
Comment (bb111189) [inline src/libDirectoryService/ViewChangePreProcessing.cpp]: return false here?

File: src/libNode/FallbackPreProcessing.cpp [hunk 3]
```diff
@@ -370,6 +388,13 @@ void Node::ComposeFallbackBlock() {
   LOG_GENERAL(INFO, "m_mediator.m_selfPeer: " << m_mediator.m_selfPeer);
   LOG_GENERAL(INFO, "LeaderNetworkInfo: " << leaderNetworkInfo);
 
+  CommitteeHash committeeHash;
+  if (!Messenger::GetShardHash(m_mediator.m_ds->m_shards.at(m_myshardId),
+                               committeeHash)) {
+    LOG_EPOCH(WARNING, to_string(m_mediator.m_currentEpochNum).c_str(),
+              "Messenger::GetShardHash failed.");
```
Comment (bb111189) [inline src/libNode/FallbackPreProcessing.cpp]: return false here?

File: src/libDirectoryService/FinalBlockPreProcessing.cpp (hunk 1 above)
Comment (kaikawaliu) [reply src/libDirectoryService/FinalBlockPreProcessing.cpp]: Okay. for now this func only returns void. Will add boolean return to interrupt the function at higher layer.

File: src/libDirectoryService/ViewChangePreProcessing.cpp (hunk 2 above)
Comment (kaikawaliu) [reply src/libDirectoryService/ViewChangePreProcessing.cpp]: Okay. for now this func only returns void. Will add boolean return to interrupt the function at higher layer.

File: src/libDirectoryService/FinalBlockPreProcessing.cpp (hunk 1 above)
Comment (kaikawaliu) [reply src/libDirectoryService/FinalBlockPreProcessing.cpp]: Fixed.

File: src/libDirectoryService/ViewChangePreProcessing.cpp (hunk 2 above)
Comment (kaikawaliu) [reply src/libDirectoryService/ViewChangePreProcessing.cpp]: Fixed.

File: src/libNode/FallbackPreProcessing.cpp (hunk 3 above)
Comment (kaikawaliu) [reply src/libNode/FallbackPreProcessing.cpp]: Fixed.

**Pull request merged**